presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -103,11 +107,17 @@ public AggregationAnalyzer(List<FieldOrExpression> groupByExpressions, Metadata
         // For a query like "SELECT * FROM T GROUP BY a", groupByExpressions will contain "a",
         // and the '*' will be expanded to Field references. Therefore we translate all simple name expressions
         // in the group by clause to fields they reference so that the expansion from '*' can be matched against them
-        for (Expression expression : Iterables.filter(expressions, instanceOf(QualifiedNameReference.class))) {
-            QualifiedName name = ((QualifiedNameReference) expression).getName();
+        for (Expression expression : Iterables.filter(expressions, columnReferences::contains)) {
+            QualifiedName name;
+            if (expression instanceof QualifiedNameReference) {
+                name = ((QualifiedNameReference) expression).getName();
+            }
+            else {
+                name = DereferenceExpression.getQualifiedName((DereferenceExpression) expression);
```

Comment:
Use `checkType` instead of this cast, so that we'll get a nice error message, if there's some how a different type of Expression here